testplan/runners/local.py

```diff
@@ -75,9 +94,20 @@ def _loop(self) -> None:
                             self,
                             exc,
                         )
-                        self._results[next_uid] = result
                     finally:
-                        self.ongoing.pop(0)
+                        with self._curr_runnable_lock:
+                            if not self._to_skip_remaining:
+                                # otherwise result from aborted test included
+                                self._results[next_uid] = result
+                                self.ongoing.pop(0)
+
+                    if thres and result.report.status <= thres:
```

Comment:
I think we can just call a method of TestRunner here to set the flag, something like

if self.cfg.stop_strategy.should_stop_plan(result):
self.parent.stop_plan()

Then TestRunner shall distribute the should_stop flag to all executors (including this one).

Executors just discard all pending_work and pretend we have finished, and triggers normal stop workflow.